gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -714,12 +726,17 @@ private void submitCompactionJobsAndWaitForCompletion() throws IOException {
    */
   private void runCompactionForDataset(Dataset dataset, boolean proceed) {
     LOG.info("Running compaction for dataset " + dataset);
-    MRCompactorJobRunner jobRunner = getMRCompactorJobRunner(dataset, dataset.priority());
-    this.jobRunnables.put(dataset, jobRunner);
-    if (proceed) {
-      jobRunner.proceed();
+
+    try {
+      MRCompactorJobRunner jobRunner = getMRCompactorJobRunner(dataset, dataset.priority());
+      this.jobRunnables.put(dataset, jobRunner);
+      if (proceed) {
+        jobRunner.proceed();
+      }
+      this.jobExecutor.execute(jobRunner);
+    } catch (Throwable t) {
```

Comment:
No need to log here, all throwables will be printed together at the end of the whole job.